common/lib/xmodule/xmodule/video_module/video_module.py

```diff
@@ -200,6 +207,25 @@ def get_html(self):
                 val_profiles = ["youtube", "desktop_webm", "desktop_mp4"]
                 val_video_urls = edxval_api.get_urls_for_profiles(self.edx_video_id, val_profiles)
 
+                # download youtube transcript for html5 video
+                if not self.sub and val_video_urls["youtube"]:
+                    #cache the transcript request
+                    cache_key = 'html5_' + val_video_urls["youtube"]
+                    html5_transcript_cache = cache.get(cache_key)
+                    if not html5_transcript_cache:
+                        try:
+                            download_youtube_subs(val_video_urls["youtube"], self, settings)
+                            item = self.runtime.modulestore.get_item(self.location)
+                            youtube_sub = val_video_urls["youtube"]
+                            item.sub = youtube_sub
+                            user = User.objects.get(id=self.system.user_id)
+                            item.save_with_metadata(user)
+                            transcript_found = True
+                        except GetTranscriptsFromYouTubeException:
+                            transcript_found = False
+
+                        cache.set(cache_key, {'youtube_hit': True, 'found': transcript_found})
```

Comment:
Shouldn't this just be `cache.set(cache_key, <bool>)`? We never inspect the contents of `html5_transcript_cache`, we just check if it's there. But, really, should we just be saving this as a field on `VideoModule`?

`html5_transcript_cache` is a misleading name. Shouldn't it be like `youtube_checked_for_transcript`? We could populate it with a datetime and "invalidate" it if it hasn't been checked in, say 5 minutes.

So the code would look more like:
```
youtube_checked_recently = datetime.now(UTC) - self.youtube_checked_for_transcript > 60 * 5
if not self.sub and val_video_urls["youtube"] and youtube_checked_recently:
    try:
        ...
    except GetTranscriptsFromYouTubeException:
        log.info("No transcript found for {video} in {course}")
```

I'm still unclear why we're saving the `youtube_sub` to `item.sub` in L220. Why don't we save `youtube_sub` to self.sub?